src/modbus-tcp-client.js

```diff
@@ -92,41 +93,40 @@ module.exports = stampit()
  
             this.log.debug('received data');
 
-            var cnt = 0;
+            buffer = Buffer.concat([buffer, data]);
 
-            while (cnt < data.length) {
+            while (buffer.length > 8) {
 
                 // 1. extract mbap
 
-                var mbap    = data.slice(cnt, cnt + 7),
-                    id      = mbap.readUInt16BE(0),
-                    len     = mbap.readUInt16BE(4);
+                var id  = buffer.readUInt16BE(0),
+                    len = buffer.readUInt16BE(4);
 
-                if (id === trashRequestId) {
```

Comment:
Why is this removed? trashed request ids is for requests that have been trashed due to timeouts or stuff. This way they can be identified.